From e3279440714433d1978cc6629dfeb87aa1a87a46 Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Tue, 2 Sep 2014 10:15:19 -0400
Subject: [net] vxlan: Enable checksum unnecessary conversions for vxlan/UDP sockets

Message-id: <1409652920-431-57-git-send-email-fwestpha@redhat.com>
Patchwork-id: 90166
O-Subject: [PATCH 56/57] [RHEL7.1 net] vxlan: Enable checksum unnecessary conversions for vxlan/UDP sockets
Bugzilla: 1131999
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: 1131999

commit c60c308cbd0e672eabaaa544216f79d8bb3e707b
Author: Tom Herbert <therbert@google.com>
Date:   Sun Aug 31 15:12:45 2014 -0700

    vxlan: Enable checksum unnecessary conversions for vxlan/UDP sockets

    Signed-off-by: Tom Herbert <therbert@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/vxlan.c b/drivers/net/vxlan.c
index 2bd00aa..73674b9 100644
--- a/drivers/net/vxlan.c
+++ b/drivers/net/vxlan.c
@@ -2336,6 +2336,8 @@ static struct socket *create_v4_sock(struct net *net, __be16 port, u32 flags)
  if (!(flags & VXLAN_F_UDP_CSUM))
   sock->sk->sk_no_check_tx = 1;
 
+ udp_set_convert_csum(sock->sk, true);
+
  return sock;
 }
 
-- 
1.7.1